test/currencyConvertor.ts

```diff
@@ -1,73 +1,175 @@
 import { ethers, waffle } from 'hardhat'
+import hre from 'hardhat';
+import '@nomiclabs/hardhat-ethers';
+import chaiAsPromised from 'chai-as-promised';
+
+
 import chai from "chai";
 import { solidity } from "ethereum-waffle";
 import CurrencyConvertorArtifact from '../artifacts/contracts/proxies/CurrencyConvertor.sol/CurrencyConvertor.json';
-import ZeroExExchangeWrapperArtifact from '../artifacts/contracts/exchange-wrappers/ZeroExExchangeWrapper.sol/ZeroExExchangeWrapper.json';
-import StarkwareArtifact from '../artifacts/contracts/exchange-wrappers/Starkware.sol/Starkware.json';
-import UsdcContractArtifact from '../artifacts/contracts/tokens/mockUsdcContract.sol/MockUsdcContract.json';
+import StarkwareArtifact from '../artifacts/contracts/mocks/Starkware.sol/MockStarkware.json';
+import ZeroExExchangeArtifact from '../artifacts/contracts/exchange-wrappers/zeroExExchangeWrapper.sol/zeroExExchangeWrapper.json';
+import { axiosRequest, encode, generateQueryPath } from './helpers';
+import { tokenAbi } from './abis';
 
-const { deployContract } = waffle
+import { CurrencyConvertor } from '../src/types';
+import { MockStarkware } from '../src/types';
+import { ZeroExExchangeWrapper } from '../src/types';
+import _ from 'underscore';
 
 
-chai.use(solidity);
+const { deployContract } = waffle
 const { expect } = chai;
+chai.use(chaiAsPromised)
+
+chai.use(solidity);
+
+const impersonatedAccount: string = '0xf0b2e1362f2381686575265799c5215ef712162f';
+const usdcAddres: string = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
+const uniswapAddress: string = '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984';
+
 describe("CurrencyConvertor", () => {
-  let currencyConvertor: any;
-  let zeroExExchangeWrapper: any;
-  let starkware: any;
-  let usdc: any;
+  let currencyConvertor: CurrencyConvertor;
+  let starkware: MockStarkware;
+  let zeroExExchangeWrapper: ZeroExExchangeWrapper;
   beforeEach(async () => {
```

Comment:
Why do you need to do this before each test, rather than before all the tests run (ie, in a `before` clause)?